libs/api/ebpf_api.cpp

```diff
@@ -1464,26 +1469,31 @@ _delete_ebpf_object(_In_opt_ _Post_invalid_ ebpf_object_t* object) noexcept
     }
 }
 
-static void
-_remove_ebpf_object_from_globals(_In_ const ebpf_object_t* object) noexcept
+_Requires_lock_not_held_(_ebpf_state_mutex) static void _remove_ebpf_object_from_globals(
+    _In_ const ebpf_object_t* object) noexcept
 {
+    std::unique_lock lock(_ebpf_state_mutex);
     EBPF_LOG_ENTRY();
     ebpf_assert(object);
     auto it = std::find(_ebpf_objects.begin(), _ebpf_objects.end(), object);
     ebpf_assert(it != _ebpf_objects.end());
     _ebpf_objects.erase(it);
 }
 
-static void
-_clean_up_ebpf_objects() noexcept
+_Requires_lock_not_held_(_ebpf_state_mutex) static void _clean_up_ebpf_objects() noexcept
 {
     EBPF_LOG_ENTRY();
-    for (auto& object : _ebpf_objects) {
+    std::unique_lock lock(_ebpf_state_mutex);
+    auto objects = std::move(_ebpf_objects);
+    _ebpf_objects.resize(0);
```

Comment:
I'm not a `std` expert at all, but I would have thought `std::move` would move everything out of `_ebpf_objects` leaving it empty, thus invalidating the need for the `.resize(0)`. No?

Reply:
Yes, I believe you are correct.